api/src/poem_backend/transactions.rs

```diff
@@ -525,12 +526,26 @@ impl TransactionsApi {
         }
         let ledger_info = self.context.get_latest_ledger_info_poem()?;
         let move_resolver = self.context.move_resolver_poem()?;
-        let (status, output) = AptosVM::simulate_signed_transaction(&txn, &move_resolver);
+        let (status, output_ext) = AptosVM::simulate_signed_transaction(&txn, &move_resolver);
         let version = ledger_info.version();
-        let exe_status = match status.into() {
-            TransactionStatus::Keep(exec_status) => exec_status,
-            _ => ExecutionStatus::MiscellaneousError(None),
+        let (output, exe_status) = match status.into() {
+            TransactionStatus::Keep(exec_status) => {
+                // Apply deltas.
+                // TODO: while `into_transaction_output_with_status()` should never fail
+                // to apply deltas, we should propagate errors properly. Fix this when
+                // VM error handling is fixed.
+                let (vm_status, output) =
+                    output_ext.into_transaction_output_with_status(&move_resolver);
+                debug_assert!(vm_status == VMStatus::Executed);
```

Comment:
this is not true, move abort/out of gas are also Keep

Reply:
move abort/out of gas are also Keep indeed, but we still apply deltas. If we have Keep(success), then deltas from txn (table length) and from epilogue (fees) are applied. If we have Keep(abort) VM starts a new session, so we only have deltas from epilogue (fees) => we apply them.

This reasoning only holds if deltas do not fail of course (which is true for V1).

Maybe having `vm_status == VMStatus::Executed` is too confusing here? Strictly speaking we do not need to have pair as an output of `into_transaction_output_with_status()`, Result<TransactionOutput> suffices

Reply:
oh I misread it as exec_status

Reply:
I guess this can be confusing then, should we make `into_transaction_output_with_status` just return result then?